src/Renci.SshNet/SftpClient.cs

```diff
@@ -2202,6 +2437,13 @@ protected override void Dispose(bool disposing)
             }
         }
 
+        private void CheckDisposedOrNotConnected()
```

Comment:
Please remove this, and perform both checks separately like to we do for the sync methods.
Also please perform the dispose check before any argument checks and the (dis)connected check afterwards.